pkgs/development/compilers/ocaml/generic.nix

```diff
@@ -73,6 +73,7 @@ stdenv.mkDerivation (args // {
   configurePlatforms = lib.optionals (lib.versionAtLeast version "4.08" && !(stdenv.isDarwin && stdenv.isAarch64)) [ "host" "target" ];
   # x86_64-unknown-linux-musl-ld: -r and -pie may not be used together
   hardeningDisable = lib.optional (lib.versionAtLeast version "4.09" && stdenv.hostPlatform.isMusl) "pie"
+    ++ lib.optional (lib.versionAtLeast version "5.0" && stdenv.cc.isClang) "strictoverflow"
```

Comment:
This line was not needed before 5.0 so the compiler changed his behaviour with respect to overflow. If they changed this it is suerely to fix a bug. I look for the reason why it was changed so we can test with nixpkgs ocaml5. So if we undone the fix at least we are aware.